### PR TITLE
Add Trace level for request events, allow logging to eventlog

### PR DESCRIPTION
**Pull request by pakrym**
Out of proc has many logs that are written during request processing that might flood debug log, add another level for them.

Allow eventlog as debug log target while I'm at it.

### CODE REVIEW

File: src/AspNetCoreModuleV2/CommonLib/EventLog.cpp [hunk 1]
```diff
@@ -5,43 +5,56 @@
...
-    static const auto versionInfoString = GetVersionInfoString();
-
-    std::array<LPCWSTR, 3> eventLogDataStrings
-    {
-        pstrMsg,
-        processIdString.c_str(),
-        versionInfoString.c_str()
-    };
-
     if (g_hEventLog != NULL)
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/EventLog.cpp]: nit: invert if.

File: src/AspNetCoreModuleV2/CommonLib/debugutil.cpp [hunk 2]
```diff
@@ -103,6 +105,7 @@ void SetDebugFlags(const std::wstring &debugValue)
             if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
             if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
             if (value >= 5) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_FILE;
+            if (value >= 6) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_EVENTLOG;
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/debugutil.cpp]: I think EVENTLOG should be above console and file.

Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/debugutil.cpp]: Why?

Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/debugutil.cpp]: Event logs are more accessible for debugging on most platforms (azure). No other big reason besides that.

Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/debugutil.cpp]: My thinking was that eventlog is the most inappropriate place to dump debug logs. Also, I would like to deprecate numbers anyway.

Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/debugutil.cpp]: Yeah let's drop the numbers.

File: test/Common.FunctionalTests/Utilities/EventLogHelpers.cs [hunk 3]
```diff
@@ -1,66 +1,65 @@
...
 
-            // Perf: only get the last 20 events from the event log.
+            // Perf: only get the last 20 events from the event log by default.
             // Eventlog is already sorted based on time of event in ascending time.
             // Add results in reverse order.
-            var expectedRegexEventLog = new Regex(expectedRegexMatchString);
             var processIdString = $"Process Id: {deploymentResult.HostProcess.Id}.";
 
-            for (var i = eventLog.Entries.Count - 1; i >= eventLog.Entries.Count - 20; i--)
+            for (var i = eventLog.Entries.Count - 1; i >= eventLog.Entries.Count - messageCount; i--)
```
Comment (jkotalik) [inline test/Common.FunctionalTests/Utilities/EventLogHelpers.cs]: Offline feedback: remove message count: https://github.com/aspnet/IISIntegration/pull/1338

**Pull request merged**